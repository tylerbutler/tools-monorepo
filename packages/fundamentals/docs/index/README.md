[**@tylerbu/fundamentals**](../README.md) â€¢ **Docs**

***

[@tylerbu/fundamentals](../modules.md) / index

# index

## References

### addAll

Re-exports [addAll](../set/functions/addAll.md)

***

### findGitRootSync

Re-exports [findGitRootSync](../git/functions/findGitRootSync.md)

***

### isSorted

Re-exports [isSorted](../array/functions/isSorted.md)
