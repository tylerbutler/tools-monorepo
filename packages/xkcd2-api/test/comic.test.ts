import { describe, expect, it } from "vitest";

import { getComicProps } from "../src/index.js";

describe("getComicProps", () => {
	it("gets existing comic", async () => {
		const comic = await getComicProps(123);
		expect(comic).toMatchSnapshot();
	});

	it("throws error when comic is not found", () => {
		expect(async () => {
			const comic = await getComicProps(0);
			expect(comic).toMatchSnapshot();
		}).not.toThrowError();
	});
});
