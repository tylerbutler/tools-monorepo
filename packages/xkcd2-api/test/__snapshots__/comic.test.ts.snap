// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getComicProps > gets existing comic 1`] = `
{
  "comic": {
    "alt": "You spin me right round, baby, right round, in a manner depriving me of an inertial reference frame.  Baby.",
    "day": "3",
    "img": "https://imgs.xkcd.com/comics/centrifugal_force.png",
    "link": "",
    "month": "7",
    "news": "",
    "num": 123,
    "safe_title": "Centrifugal Force",
    "title": "Centrifugal Force",
    "transcript": "[[ Bond is tied to a giant centrifuge ]]
Hat Guy: Do you like my centrifuge, Mister Bond? When I throw the lever, you will feel centrifugal force crush every bone in your body.
Mr. Bond: You mean centripetal force. There's no such thing as centrifugal force.
Hat Guy: A laughable claim, Mister Bond, perpetuated by overzealous teachers of science. Simply construct Newton's laws into a rotating system and you will see a centrifugal force term appear as plain as day.
Mr. Bond: Come now, do you really expect me to do coordinate substitution in my head while strapped to a centrifuge?
Hat Guy: No, Mr. Bond. I expect you to die.
{{ alt: You spin me right round, baby, right round, in a manner depriving me of an inertial reference frame.  Baby. }}",
    "year": "2006",
  },
  "nextId": "124",
  "previousId": "122",
}
`;
